File: DataManager.js
import fs from 'fs';
import Logger from "./logger.js";

export class DataManager {
  constructor() {
    this.filename = 'latest_price_db.json';
    this.data = this.init();
  }

  init() {
    try {
      const fileContent = fs.readFileSync(this.filename, 'utf8');
      return JSON.parse(fileContent);
    } catch (error) {
      if (error.code === 'ENOENT') {
        return {};
      } else {
        Logger.error('Error initializing DataManager:', error);
        throw error;
      }
    }
  }

  async save() {
    try {
      await fs.promises.writeFile(this.filename, JSON.stringify(this.data, null, 2));
    } catch (error) {
      Logger.error('Error saving data:', error);
      throw error;
    }
  }

  getValue(key) {
    return this.data[key];
  }

  async setMultipleValues(keyValuePairs) {
    for (const [key, value] of keyValuePairs) {
      this.data[key] = value;
    }
    await this.save();
  }
}

File: data_manager.py
import json
from logger import Logger


class DataManager:
    def __init__(self):
        self.filename = 'latest_price_db.json'
        self.data = self.init()

    def init(self):
        try:
            with open(self.filename, 'r') as file:
                return json.load(file)
        except FileNotFoundError:
            return {}
        except json.JSONDecodeError as error:
            Logger.error('Error initializing DataManager:', error)
            raise

    def save(self):
        try:
            with open(self.filename, 'w') as file:
                json.dump(self.data, file, indent=2)
        except IOError as error:
            Logger.error('Error saving data:', error)
            raise

    def get_value(self, key):
        return self.data.get(key)

    def set_multiple_values(self, key_value_pairs):
        for key, value in key_value_pairs:
            self.data[key] = value
        self.save()
